### DISCUSSION TITLE
The output maxdets is different from pycocotools.

### DISCUSSION THREAD

**Discussion by DiYaGo-zzh** [open-mmlab/mmdetection#9213] [Q&A] [answered]
I ran into some problems about CoCo format Dataset train or test(test directives is 'python tools/test.py configs/libra_rcnn/libra_faster_rcnn_r50_fpn_1x_coco.py work_dirs/libra_faster_rcnn_r50_fpn_1x_coco/latest.pth --out result/result.pkl --eval bbox'),I noticed that the master version output the CoCo indicator like figure when i train model :
![图片](https://user-images.githubusercontent.com/77278409/199643230-2f60bd6a-bd4b-44fa-9c3b-e556d95a41fb.png)
but i want to align with pycocotools,That is, change the maxDets to [1,10,100] , so i revise the mmdet/datasets/coco.py file line 486 to change cocoEval.params.maxDets like figure:
![图片](https://user-images.githubusercontent.com/77278409/199643451-54d4a155-0b5e-4192-a9fb-79a9ec3f4236.png)
but after revise code and rerun the train or test ,it still output the previous format,if i want to align with pycocotools ,shuld i revise other files?

**Comment by DiYaGo-zzh** [comment, marked as answer]
Looks like I've found a temporary workaround,when use this directives('python tools/test.py configs/libra_rcnn/libra_faster_rcnn_r50_fpn_1x_coco.py work_dirs/libra_faster_rcnn_r50_fpn_1x_coco/latest.pth --out result/result.pkl --eval bbox --eval-options "proposal_nums=(1, 10, 100)"') to test ,and it work,like figure,
![图片](https://user-images.githubusercontent.com/77278409/199647907-a35377d4-740a-426f-8f51-e575d4c4b199.png)
But when i use previous instructions to test and output like this,
![图片](https://user-images.githubusercontent.com/77278409/199648060-1123030f-0dc8-4ec5-ab6a-c543a048f42c.png)
So you can see the values of these indicators have hardly changed,Apart from this(in red line box):
![图片](https://user-images.githubusercontent.com/77278409/199648277-3a31183c-08b4-4976-a157-29ffe1ccb49e.png)
So I'm not quite sure if this is really a valid modification，or if it just changes the print format of the output, and doesn't actually pass the parameters into cocoEval.

**Reply by BIGWangYuDong** [reply]
Have you checked other repos settings? such as Detectron2?

**Reply by DiYaGo-zzh** [reply]
Not yet , At first I just wanted to use mmdetection evaluates some baselines (in my datasets) for comparative experiments in the paper and found that maxdets and pycocotools standard output seem to be different. And I'm not sure if the evaluation results obtained by passing this parameter are correct. I may try other repos to compare the results in the future.

**Comment by BIGWangYuDong** [comment]
This is a good question. Recall is always used for RPN, this setting will not affect mAP in COCO dataset. Hence, we will going to check other repos, such as Detectron2. This setting should be unified with other repos.

**Reply by DiYaGo-zzh** [reply]
So can I simply understand that maxdets will not affect the AP in coco format dataset, only change the AR?
